DEVELOPMENT.md

```diff
@@ -187,28 +144,87 @@ git remote add upstream git@github.com:tektoncd/pipeline.git
 git remote set-url --push upstream no_push
 ```
 
-_Adding the `upstream` remote sets you up nicely for regularly
-[syncing your fork](https://help.github.com/articles/syncing-a-fork/)._
+> **Note** Adding the `upstream` remote to your `.git/config` sets you up nicely for regularly [syncing your fork](https://help.github.com/articles/syncing-a-fork/) and submitting pull requests.
 
-## Kubernetes cluster
+### Configure Container Registry
 
-The recommended configuration is:
+Depending on your chosen container registry that you set in the `KO_DOCKER_REPO` environment variable, you may need to additionally configure access control to allow `ko` to authenticate to it.
 
--   Kubernetes version 1.17 or later
--   4 vCPU nodes (`n1-standard-4`)
--   Node autoscaling, up to 3 nodes
--   API scopes for cloud-platform
+#### Using Google Container Registry (GCR)
 
+If using GCR with `ko`, make sure to configure
+[authentication](https://cloud.google.com/container-registry/docs/advanced-authentication#standalone_docker_credential_helper)
+for your `KO_DOCKER_REPO` if required. To be able to push images to
+`gcr.io/<project>`, you need to run this once:
+
+```shell
+gcloud auth configure-docker
+```
+
+Besides, if your registry `KO_DOCKER_REPO` is private, then you also should create a [secret](https://kubernetes.io/docs/concepts/configuration/secret/#docker-config-secrets)
+and add it to `tekton-pipelines-controller` and `tekton-pipelines-webhook` serviceAccounts accordingly to pull images.
+
+1. Create a secret
+    ```yaml
+    kubectl create secret docker-registry ${SECRET_NAME} \
+    --docker-username=${USERNAME} \
+    --docker-password=${PASSWORD} \
+    --docker-email=me@here.com \
+    --namespace=tekton-pipelines
+    ```
+
+2. Add the `secret` to the K8s `ServiceAccount` configuration file
+
+    Before using `ko` to deploy `tekton-pipelines`, you need to change [the contents of the ServiceAccount](./config/200-serviceaccount.yaml) configuration file as shown below:
+
+    ```yaml
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+    name: tekton-pipelines-controller
+    namespace: tekton-pipelines
+    labels:
+        app.kubernetes.io/component: controller
+        app.kubernetes.io/instance: default
+        app.kubernetes.io/part-of: tekton-pipelines
+    imagePullSecrets:
+    - name: ${SECRET_NAME}
+    ---
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+    name: tekton-pipelines-webhook
+    namespace: tekton-pipelines
+    labels:
+        app.kubernetes.io/component: webhook
+        app.kubernetes.io/instance: default
+        app.kubernetes.io/part-of: tekton-pipelines
+    imagePullSecrets:
+    - name: ${SECRET_NAME}
+    ```
+
+---
+
+## Building and deploying
+
+### Setup a Kubernetes cluster
+
+The recommended configuration is:
+
+-  Kubernetes version 1.17 or later
+-  4 vCPU nodes (`n1-standard-4`)
```

Comment:
I wonder if we should be pointing developers at Kind, instead of cloud resources?  Might be a lower barrier given it's free?

If we stick with machine recommendations, maybe change to `e2-standard-2`?  It's cheaper and, at least during my development, Tekton Pipelines alone hasn't required lots of CPU / RAM resources.

Reply:
Ah, I also notice we have GKE-specific instructions below.  Maybe move the specific machine recommendation (which is a Google Cloud designation I think?) to that section instead?

Reply:
> Ah, I also notice we have GKE-specific instructions below. Maybe move the specific machine recommendation (which is a Google Cloud designation I think?) to that section instead?

yes, I agree that referencing pre-packaged, named tiers for specific providers should not be part of the canonical documentation.   Not sure when `n1-standard-4` was added, but likely should remove altogether.

Reply:
yup, I agree, these recommendations might confuse developers compared to the machine specific recommendations. @mrutkows please feel free to delete these from here or can be addressed in a follow up PR.